pkg/deps/annotation_matcher.go

```diff
@@ -75,9 +81,12 @@ func AnnotationMatcher(criteria *AnnotationCriteria) Matcher {
 		}
 
 		matchesAnnot := func(key string, vals []string, annots map[string]string) bool {
+			if _, ok := annots[key]; ok && len(vals) == 0 {
+				return true
+			}
 			matchesAny := false
 			for _, val := range vals {
-				if m.Annotations[key] == val {
+				if annots[key] == val {
 					matchesAny = true
```

Comment:
you return early in the above statement, but here you set a value and return at the end.  One way or the other better.

Reply:
Done